circuit-retained/src/androidMain/kotlin/com/slack/circuit/retained/AndroidContinuity.kt

```diff
@@ -18,11 +18,30 @@ import androidx.lifecycle.compose.LifecycleStartEffect
 import androidx.lifecycle.viewmodel.CreationExtras
 import androidx.lifecycle.viewmodel.compose.viewModel
 
-internal class ContinuityViewModel : ViewModel(), RetainedStateRegistry, CanRetainChecker {
+/**
+ * A [RetainedStateRegistry] used by [continuityRetainedStateRegistry] that can update its
+ * [CanRetainChecker].
+ */
+public interface ContinuityRetainedStateRegistry : RetainedStateRegistry {
```

Comment:
What do you think of us just dropping `Continuity` from this naming and maybe calling it `UpdatableRetainedStateRegistry`? Then the below one could be a `ViewModelRetainedStateRegistryFactory` too. This name was really just sort of a placeholder as it took inspiration from @rjrjr's droidcon nyc talk about the subject way back in ~2023 (2022?)

Reply:
Already planning on following up with some more [changes](https://github.com/slackhq/circuit/compare/main...j-retained-always) for this 🚀  
Its interesting with "continuity" being the root registry to handle any platform specific oddities. Right now its mostly just an alias for `rememberRetainedStateRegistry`, except on android where we want it to follow a different lifecycle 🤔